apps/console/src/features/sms-providers/pages/sms-providers.tsx

```diff
@@ -432,39 +441,47 @@ const SMSProviders: FunctionComponent<SMSProviderPageInterface> = (
         return error;
     };
 
-    const handleConfigurationDelete = async (deleteBeforeUpdate?: boolean): Promise<boolean> => {
-        !deleteBeforeUpdate && setIsDeleting(true);
+    const handleConfigurationDelete = async (): Promise<boolean> => {
+        setIsDeleting(true);
 
         return deleteSMSProviders()
             .then(() => {
-                !deleteBeforeUpdate && handleDeleteSuccess();
-                !deleteBeforeUpdate && setSmsProviderSettings({ providerParams: {}, 
+                handleDeleteSuccess();
+                setSmsProviderSettings({ providerParams: {}, 
                     selectedProvider: SMSProviderConstants.TWILIO_SMS_PROVIDER });
-                !deleteBeforeUpdate && mutateSMSProviderConfig();
+                mutateSMSProviderConfig();
 
                 return true;
             })
             .catch((e: IdentityAppsApiException) => {
-                !deleteBeforeUpdate && handleDeleteError();
-                if (deleteBeforeUpdate && e.response.status === 404) {
-                    return true;
-                }
-
+                handleDeleteError(e);
+                
                 return false;
             }).finally(() => {
                 setIsDeleting(false);
             });
     };
 
 
-    const handleDeleteError = () => {
+    const handleDeleteError = (error?: IdentityAppsApiException) => {
+        let errorMessage: string = t("extensions:develop.smsProviders." +
+        "notifications.deleteConfiguration.error.message");
+
+        let errorDescription: string = t("extensions:develop.smsProviders." +
+        "notifications.deleteConfiguration.error.description");
+
+        // If the SMS provider is being used by SMS OTP connection and it is
+        // configured for one or more applications
+        if (error?.response?.data?.code === "NSM-60008") {
```

Comment:
Move the error code to the feature's constants file and refer from here.

Reply:
Fixed with e90387f